packages/react-sdk/src/components/elements/image/ImageDisplay.tsx

```diff
@@ -64,132 +157,21 @@ function ImageDisplay({
+    { widgetApi, baseUrl, mxc, mimeType },
+    downloadFile,
+    { suspense: true },
+  );
 
   const handleLoad = useCallback(() => {
-    setLoading(false);
-    setLoadError(false);
-
     // This can happen directly when the image is loaded and saves some memory.
     if (imageUri) {
       URL.revokeObjectURL(imageUri);
     }
-  }, [setLoading, setLoadError, imageUri]);
-
-  const handleLoadError = useCallback(() => {
-    setLoading(false);
-    setLoadError(true);
-  }, [setLoading, setLoadError]);
-
-  useEffect(() => {
-    const downloadFile = async () => {
-      try {
-        const result = await tryDownloadFileWithWidgetApi(mxc);
-        const blob = getBlobFromResult(result, mimeType);
-        const downloadedFileDataUrl = createObjectUrlFromBlob(blob);
-        setImageUri(downloadedFileDataUrl);
-      } catch (error) {
-        handleDownloadError(error as Error);
-      }
-    };
-
-    const tryDownloadFileWithWidgetApi = async (mxc: string) => {
-      try {
-        const result = await widgetApi.downloadFile(mxc);
-        return result;
-      } catch {
-        throw new WidgetApiActionError('downloadFile not available');
-      }
-    };
-
-    const getBlobFromResult = (
-      result: IDownloadFileActionFromWidgetResponseData,
-      mimeType: string,
-    ): Blob => {
-      if (!(result.file instanceof Blob)) {
-        throw new Error('Got non Blob file response');
-      }
-      return result.file.slice(0, result.file.size, mimeType);
-    };
-
-    const createObjectUrlFromBlob = (blob: Blob): string => {
-      const url = URL.createObjectURL(blob);
-      if (url === '') {
-        throw new Error('Failed to create object URL');
-      }
-      return url;
-    };
-
-    const handleDownloadError = (error: Error) => {
-      if (error instanceof WidgetApiActionError) {
-        tryFallbackDownload();
-      } else {
-        setLoadError(true);
-      }
-    };
-
-    const tryFallbackDownload = async () => {
-      let abortController: AbortController | undefined;
-
-      const httpUrl = convertMxcToHttpUrl(mxc, baseUrl);
-
-      if (httpUrl === null) {
-        setImageUri('');
-        return;
-      }
-
-      if (mimeType === 'image/svg+xml') {
-        abortController = new AbortController();
-        try {
-          const response = await fetch(httpUrl, {
-            signal: abortController.signal,
-          });
-          const rawBlob = await response.blob();
-          const svgBlob = rawBlob.slice(0, rawBlob.size, mimeType);
-          setImageUri(URL.createObjectURL(svgBlob));
-        } catch (fetchError) {
-          console.error('Failed to fetch SVG image:', fetchError);
-          setLoadError(true);
-        }
-        return;
-      }
-
-      setImageUri(httpUrl);
-
-      return () => {
-        if (abortController) {
-          abortController.abort();
-        }
-      };
-    };
-
-    downloadFile();
-  }, [baseUrl, mimeType, mxc, widgetApi]);
-
-  const renderedSkeleton =
```

Comment:
The eerrors and skeleton are now handled one component above with error handling and suspense